docs/design/software_architecture/index.md

```diff
@@ -4,59 +4,54 @@ This section explains an overview of software architecture.
 
 ![architecture](../../imgs/architecture.drawio.png)
 
-CARET is divided into three phases; recording phase, configuration phase and analyzing phase.
+CARET serves three phases; recording, configuration and visualization.
 
-## Recording Phase
+## Recording
 
-In the recording phase, CARET records system execution data from tracepoints.
+In the recording phase, CARET records events data obtained from tracepoints during application's runtime.
 
-In addition to the tracepoints built into ROS2, CARET adds tracepoints by several methods; CARET_trace CARET_rclcpp and TILDE.
-For flexibility in adding tracepoints, tracepoints are added by hooks as possible.
-Only if it is not possible to add tracepoints by hooks, tracepoints are added by other methods as a supplementary method.
+CARET adopts LTTng as a tracing mechanism. LTTng session daemon collects events from tracepoints embedded in a target application.
+`rclcpp` provided by ROS 2 has original tracepoints and CARET uses some of them also. CARET collects its dedicated tracepoints added by function hooking.
+To add tracepoints flexibly, function hooking is .
```

Comment:
```suggestion
Function hooks are actively used in CARET because they do not require changes to multiple existing packages and allow adding trace points in a flexible way.
```

Reply:
I'd like to align `function hooking`, instead of `hooks` or `function hooks`.